dashboard-all-the-things/src/app/dashboard-carousel/dashboard-carousel.component.ts

```diff
@@ -1,51 +1,64 @@
-import { Component, OnInit, OnDestroy, ElementRef } from '@angular/core';
-import { Subscription } from 'rxjs';
-import { Observable } from 'rxjs/Observable';
+import { ExternalUrlComponent } from '../dashboards/external-url/external-url.component';
+import { ReviewComponent } from '../dashboards/review';
+import { SonarCoverageComponent } from '../dashboards/sonar-coverage';
+import { StackOverflowComponent } from '../dashboards/stackoverflow/stackoverflow.component';
+import { WidgetEvent } from '../dashboards/WidgetEvent';
+import { DynamicComponent } from './dynamic.component';
+import { SocketService } from './socket.service';
+import { Component, OnDestroy, OnInit, ViewChild } from '@angular/core';
+import { ConnectableObservable, Subscription } from 'rxjs';
+
+let COMPONENTS = [
+  { key: "coverage", type: SonarCoverageComponent },
+  { key: "ciWall", type: ExternalUrlComponent },
+  { key: "reviews", type: ReviewComponent },
+  { key: "stackoverflow", type: StackOverflowComponent }
+];
 
 @Component({
   selector: 'app-dashboard-carousel',
   templateUrl: 'dashboard-carousel.component.html',
-  styleUrls: ['dashboard-carousel.component.scss']
+  styleUrls: ['dashboard-carousel.component.scss'],
+  entryComponents: COMPONENTS.map((el) => el.type)
+    .filter((elem, index, arr) => arr.indexOf(elem) === index)
```

Comment:
For now yes.
In the future the idea is to have multiple widgets rendered by the same component so we need to make sure to have a list with only unique component.
Idea would be that the server would sent a payload that might contains a mix of texts and images and will ask for a template to render those infos, the client shouldn't be aware if the message is about sonar or slack but just on how the infos should be rendered.
Anyway this might be YAGNI then for now :)